### PR TITLE
Remove support for Node 14, raise minimum version to 16

### PR DESCRIPTION
**Pull request by levibuzolic**
As discussed in #450, this raises the minimum node version support to 16 to fix some compatibility issues with the latest version of various dependencies.

https://github.com/oblador/loki/pull/450#discussion_r1167658000

### CODE REVIEW

**Comment by oblador** [comment]
I suspect the failing integration test is due to the SB 7 release that somehow broke stuff. Does it pass on master for you?

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v14.17.3
+v16.20.0
```
Comment (techeverri) [inline .nvmrc:1]: We should also update this one 👇 

https://github.com/oblador/loki/blob/7ca6aa0101151dd7ec56652708cdafd1083079cf/docs/serverless.md?plain=1#L66

Comment (techeverri) [reply .nvmrc:1]: I'd also suggest adding " (Node.js v16.x or newer)" to the "prerequisites" sections

https://github.com/oblador/loki/blob/7ca6aa0101151dd7ec56652708cdafd1083079cf/packages/loki/README.md?plain=1#L25

https://github.com/oblador/loki/blob/7ca6aa0101151dd7ec56652708cdafd1083079cf/docs/getting-started.md?plain=1#L8

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "publish-docs": "GIT_USER=oblador yarn workspace @loki/website publish-gh-pages"
   },
   "engines": {
-    "node": ">=14.0.0"
+    "node": ">=16.0.0"
```
Comment (techeverri) [inline package.json:18]:
```suggestion
    "node": ">=16"
```

Comment (techeverri) [reply package.json:18]: I used https://semver.npmjs.com/ to compare the matches
_Package: `node`, Version range: `>=16`_

Comment (techeverri) [reply package.json:18]: Shouldn't we have at least the same version that we have in the `.nvmrc`?

```suggestion
    "node": ">=16.20.0"
```

Comment (levibuzolic) [reply package.json:18]: @techeverri the `.nvmrc` is for development, but the `engines` is for consumers. There's no strict reason they need to be identical.

**Comment by levibuzolic** [comment]
> @oblador I suspect the failing integration test is due to the SB 7 release that somehow broke stuff. Does it pass on master for you?

`master` doesn't pass locally anymore because Storybook 7 also requires Node 16 now.

**Comment by levibuzolic** [comment]
@oblador Digging into the failure a little bit more, it looks Storybook 7 requires a web server because they've switched to usedusing JavaScript modules (`.mjs`) which can't be loaded from the local filesystem.

I quickly hacked in a `http-server` wrapper around the static build (though we might as well just run the storybook server directly) which gets a little further, however it looks like loki [isn't compatible with Storybook 7](https://github.com/oblador/loki/issues/436), so that'll need to be fixed first.

**Comment by oblador** [comment]
@levibuzolic 0.31.2 is out now with storybook 7 fixes, could you please give it another try? 🙏

**Comment by levibuzolic** [comment]
@oblador should be good to go now.

**Review by oblador — APPROVED** [review]

**Comment by oblador** [comment]
Thanks!

**Pull request merged**